examples/plugins/ebpf/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+func main() {
+	hostname, _ := os.Hostname()
+	var (
+		addr   = flag.String("addr", "/var/run/scope/plugins/ebpf.sock", "unix socket to listen for connections on")
+		hostID = flag.String("hostname", hostname, "hostname of the host running this plugin")
+	)
+	flag.Parse()
+
+	log.Println("Starting...")
+
+	// Compile and run the ebpf script
+	ebpf := exec.Command("http-parse-simple.py")
+	ebpf.Stderr = os.Stderr
+	stdout, err := ebpf.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := ebpf.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if ebpf.Process != nil {
+			ebpf.Process.Kill()
+		}
+	}()
+
+	os.Remove(*addr)
```

Comment:
CNI just execs its plugins, rather than using any kind of socket arrangement. This works pretty well because plugins are invoked to perform operations, and state is out-of-band.

Docker plugins use HTTP over a socket to a long-running process. A downside of this is that someone (probably the user) now has to manage the lifecycle of the long-running process. But it seems appropriate enough if that process is collecting data that the system will occasionally request.